File: app/src/main/java/com/example/android_storagepart2/ViewModel.java
package com.example.android_storagepart2;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class ViewModel extends AndroidViewModel {
    private Employee_Repository repository;
    private LiveData<List<Employee>> allEmployees;

    public ViewModel(@NonNull Application application) {
        super(application);
        repository = new Employee_Repository(application);
        allEmployees = repository.getAllEmployees();
    }

     void insert(Employee employee) {
        repository.insert(employee);
    }

     void update(Employee employee) {
        repository.update(employee);
    }

     void delete(Employee employee) {
        repository.delete(employee);
    }

     LiveData<List<Employee>> getAllEmployees() {
        return allEmployees;
    }
}

File: exercise_14_android_introduction_to_jetpack_component_part_2/src/main/java/com/example/exercise_14_android_introduction_to_jetpack_component_part_2/viewmodel/ViewModel.kt
package com.example.exercise_14_android_introduction_to_jetpack_component_part_2.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.example.exercise_14_android_introduction_to_jetpack_component_part_2.database.UserRepository
import com.example.exercise_14_android_introduction_to_jetpack_component_part_2.model.User

/**
 * ViewModel class
 */
class ViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: UserRepository = UserRepository(application)


    val allUsers: LiveData<List<User>>

    /**
     * Insert user
     */
    fun insert(user: User?) {
        repository.insert(user)
    }

    /**
     * update user
     */
    fun update(user: User?) {
        repository.update(user)
    }

    /**
     * delete user
     */
    fun delete(user: User?) {
        repository.delete(user)
    }

    /**
     * return allUsers from repo
     */
    init {
        allUsers = repository.allUsers
    }
}